xhr.bs

```diff
@@ -1609,7 +1609,7 @@ typedef (File or USVString) FormDataEntryValue;
 
 [Exposed=(Window,Worker)]
 interface FormData {
-  constructor(optional HTMLFormElement form);
+  constructor(optional HTMLFormElement form, optional HTMLElement? submitter = null);
```

Comment:
Found a nit after all. Is there a good reason to allow null here? Or would requiring undefined by fine? If there's no use case for passing null we should not allow it and change the language below from "is non-null" to "is given".

(And then have an additional test that null throws.)

Reply:
I don't have a strong preference here, mainly I was making it consistent with [HTMLFormElement.requestSubmit](https://html.spec.whatwg.org/multipage/forms.html#dom-form-requestsubmit()) which allows a null `submitter`.

Reply:
Okay, that's fairly compelling I suppose, although I don't remember the rationale for that one. 😅

Let's leave this as-is then unless @domenic cares more strongly than I do.

Reply:
It's possible it was an oversight, as you had a similar suggestion when [requestSubmit was proposed](https://github.com/whatwg/html/issues/4187#issuecomment-486228427) and afaict it was never directly addressed

Reply:
Ah here's a good justification for `requestSubmit` accepting null: https://github.com/whatwg/html/issues/3195#issuecomment-547475867

Similarly this could be useful for the `FormData` constructor, e.g. `SubmitEvent.submitter` could be null, and it would be nice to avoid null checks:

```javascript
myForm.addEventListener("submit", () => {
  const formData = new FormData(e.target, e.submitter);
  // ...
});
```